### PR TITLE
Add opa to kube poc

### PR DESCRIPTION
**Pull request by sidpremkumar**
## Description of the change

> Description here (why is it needed, what does it do)

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Refactor (code change with no functionality change)
- [ ] Enhancement (minor change below the level of a feature)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Remove feature (removes a feature we don't support anymore)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related JIRA tickets

Relates to JIRA: CWC-XXX

## Relevant release note information

Release Notes:

## Potential Security Impacts

Does this PR have any security impact?
Does it fix a security issue?
Does it add attack surface?
Why do we think this change is safe?

### Development

- [ ] Does the code changed/added as part of this pull request have test coverage?
- [ ] Do all tests related to the changed code pass in development?
- [ ] Have you ensured that at least one other person has tested your code?
- [ ] Have you tested the code?
- [ ] Have you attached any pictures (if relevant)?

### CODE REVIEW

File: src/cli-driver.ts [hunk 1]
```diff
@@ -73,8 +80,9 @@ export class CliDriver
     {
         // ref: https://nodejs.org/api/process.html#process_process_argv0
         this.processName = process.argv0;
-
-        yargs(process.argv.slice(2))
+        
+        // @ts-ignore TS2589
```
[the remark this answers is not in the document]
Comment (sidpremkumar) [reply src/cli-driver.ts:84]: bc our args object has become too ambiguous,

[the remark this answers is not in the document]
Comment (sidpremkumar) [reply src/cli-driver.ts:84]: https://github.com/microsoft/TypeScript/issues/34933

File: src/handlers/remove-user.handler.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    var policies = await policyService.ListAllPolicies();
+    
+    // Loop till we find the one we are looking for
+    for (var policy of policies) {
+        if (policy.name == clusterName) {
+            // Then remove the user from the policy
+            var newSubjects = []
+            for (var subject of policy.subjects) {
+                if (subject.id != userInfo.id) {
+                    newSubjects.push(subject)
```
[the remark this answers is not in the document]
Comment (sidpremkumar) [reply src/handlers/remove-user.handler.ts:33]: left a todo

File: src/handlers/start-kube-daemon.handler.ts [hunk 3]
```diff
@@ -1,91 +1,153 @@
...
+    if (response.allowed != true) {
+        logger.error(`You do not have the correct policy setup to access ${assumeCluster} as ${assumeUser}`);
+        await cleanExit(1, logger);
+    }
 
-export async function startKubeDaemonHandler(connectUser: string, connectCluster: string, configService: ConfigService, logger: Logger) {
-    // First check if we've already started a process
+    // Check if we've already started a process
     var kubeConfig = configService.getKubeConfig();
     // TODO : Make sure the user has created a kubeConfig before
```
[the remark this answers is not in the document]
Comment (sidpremkumar) [reply src/handlers/start-kube-daemon.handler.ts:33]: added

File: src/handlers/start-kube-daemon.handler.ts [hunk 4]
```diff
@@ -1,91 +1,153 @@
...
+    // Copy the file to the computers file system
+    await copy(daemonExecPath, finalDaemonPath); // this should work
 
-async function generateUniqueId(): Promise<number> {
-    // Helper function to generate a uniqueId
-    // ret number: unique number
-    return Math.round(Math.random() * 1000);
-}
+    // Grant execute permission
+    await chmod(finalDaemonPath, 0o765);
```
[the remark this answers is not in the document]
Comment (sidpremkumar) [reply src/handlers/start-kube-daemon.handler.ts:149]: added a todo

File: src/handlers/status.handler.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import { Logger } from '../logger.service/logger';
+import { ConfigService } from '../config.service/config.service';
+import { cleanExit } from './clean-exit.handler';
+
+export async function statusHandler(
+    configService: ConfigService,
+    logger: Logger
+) {
+    // First get the status from the config service
+    var kubeConfig = configService.getKubeConfig();
```
[the remark this answers is not in the document]
Comment (sidpremkumar) [reply src/handlers/status.handler.ts:10]: we change it  on line 18 if the daemon quite unexpectedly

**Review by tflpd — APPROVED** [review]

**Pull request merged**